packages/components/src/components/input-date/input-date.e2e.ts

```diff
@@ -36,6 +37,32 @@ test.describe('kol-input-date', () => {
 			}, TEST_DATE);
 			await expect(page.locator('input')).toHaveValue('04:02');
 		});
+		test('should return Date when initial value is a Date object and trigger onChange', async ({ page }) => {
+			await page.setContent('<kol-input-date _label="Date input"></kol-input-date>');
+			await page.locator('kol-input-date').evaluate((element: HTMLKolInputDateElement, date) => {
+				element._value = date;
+				element.addEventListener('change', (e: Event) => {
+					const value: Date | string = (e as CustomEvent).detail.value;
+					expect(value).toBeInstanceOf(Date);
+					if (value instanceof Date) expect(value.toISOString()).toBe('2021-04-04T00:00:00.000Z');
+				});
+			}, TEST_DATE);
+			await page.locator('input').fill('2021-04-04');
+			await page.locator('input').dispatchEvent('change');
+		});
+
+		test('should return the correct value with getValue()', async ({ page }) => {
+			await page.setContent('<kol-input-date _label="Date input"></kol-input-date>');
+			await page.locator('kol-input-date').evaluate((element: HTMLKolInputDateElement, date) => {
+				element._value = date;
+			}, TEST_DATE);
+			await page.locator('input').fill('2021-04-04');
+			const value = await page.locator('kol-input-date').evaluate((element: HTMLKolInputDateElement) => {
+				return element.getValue();
+			});
+			expect(value).toBeInstanceOf(Date);
+			if (value instanceof Date) expect(value?.toISOString()).toBe('2021-04-04T00:00:00.000Z');
```

Comment:
Expectations mit Bedingung würde ich auch gerne ganz vermeiden, um die Tests simpel zu halten.

In diesem Fall ist die Bedingung auch gar nicht notwendig, weil wir den Type bereits in der Expectation davor testen. Sollte `value` einen anderen Type als `Date` haben, schlägt der Test bereits fehl.